### PR TITLE
SMV: set pretty names for top-level identifiers

### PR DESCRIPTION
**Pull request by kroening**
Identifiers in module main now no longer carry the prefix `main::var::`.

### CODE REVIEW

**Comment by tautschnig** [comment]
Rebase required to resolve conflicts.

**Comment by kroening** [comment]
done

File: src/smvlang/expr2smv.h [hunk 1]
```diff
@@ -8,6 +8,5 @@ Author: Daniel Kroening, kroening@kroening.com
 
 #include <util/expr.h>
 
-bool expr2smv(const exprt &expr, std::string &code);
-bool type2smv(const typet &type, std::string &code);
-
+bool expr2smv(const exprt &, std::string &code, const namespacet &);
+bool type2smv(const typet &, std::string &code, const namespacet &);
```
Comment (tautschnig) [inline src/smvlang/expr2smv.h:12]: Not for this PR, but rather asking about potential future work: Is `bool` really the appropriate return type? Shouldn't places that `return true` really be using `UNIMPLEMENTED` instead? Then the return type could instead be `std::string`.

File: src/smvlang/smv_language.cpp [hunk 2]
```diff
@@ -168,7 +172,9 @@ void smv_languaget::show_parse(std::ostream &out, message_handlert &)
       if(it->second.type.id()=="submodule")
       {
         std::string msg;
-        type2smv(it->second.type, msg);
+        symbol_tablet symbol_table;
+        namespacet ns(symbol_table);
```
Comment (tautschnig) [inline src/smvlang/smv_language.cpp:176]: Do we really not have a symbol table or namespace available when invoking `show_parse`?

**Review by tautschnig — APPROVED** [review]

File: src/smvlang/expr2smv.h (hunk 1 above)
Comment (kroening) [reply src/smvlang/expr2smv.h:12]: Another option would be to send anything that that code doesn't know how to format to `format_expr`.

File: src/smvlang/smv_language.cpp (hunk 2 above)
Comment (kroening) [reply src/smvlang/smv_language.cpp:176]: Yes, this is pre typechecking.

**Pull request merged**